### DISCUSSION TITLE
How could I compute the powers of adjacency matrix efficiently in PyG?

### DISCUSSION THREAD

**Discussion by Kousaka-Honoka** [pyg-team/pytorch_geometric#4557] [Q&A] [answered]
I didn't see the multiplication of sparse tensor in pytorch document, so are there any approaches in PyG that could I implement this operation?
AND I used torch.sparse.mm() to compute  A@A, but it gives me a ridiculous bug which is **_'SparseTensor' object has no attribute 'is_sparse'_**

**Comment by Kousaka-Honoka** [comment, marked as answer]
Well, the **_SparseTensor_** in **_torch_sparse_** is different from **_torch._sparse.tensor__**, and I found It supports @ operator, which can compute the powers of adjacency matrix iteratively

**Reply by rusty1s** [reply]
Yes, the `SparseTensor` of `torch-sparse` supports sparse-sparse matrix multiplication via `@`. Keep in mind that this will increase the number of non-zero elements in your graph by a large margin :)
